### PR TITLE
UNDERTOW-2093 detailed followup.

### PR DESCRIPTION
**Pull request by baranowb**

### CODE REVIEW

File: spotbugs-exclude.xml [hunk 1]
```diff
@@ -698,6 +698,45 @@
             <And>
                 <Class name="io.undertow.servlet.core.SessionListenerBridge"/>
                 <Method name="sessionDestroyed"/>
+            </And>
+              <And>
+                <!-- TODO: figure out lambda ref? -->
```
Comment (fl4via) [inline spotbugs-exclude.xml:703]: This comment seems to be unrelated to throws runnable rule

File: core/src/main/java/io/undertow/security/impl/SingleSignOnAuthenticationMechanism.java [hunk 2]
```diff
@@ -120,6 +121,8 @@ public void handleNotification(SecurityNotification notification) {
                     });
                     log.tracef("Authenticated account %s using SSO", verified.getPrincipal().getName());
                     return AuthenticationMechanismOutcome.AUTHENTICATED;
+                } else {
```
Comment (fl4via) [inline core/src/main/java/io/undertow/security/impl/SingleSignOnAuthenticationMechanism.java:124]: probably something is missing here?

**Review by fl4via — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/undertow/security/impl/SingleSignOnAuthenticationMechanism.java (hunk 2 above)
Comment (baranowb) [reply core/src/main/java/io/undertow/security/impl/SingleSignOnAuthenticationMechanism.java:124]: AFAIR no

File: spotbugs-exclude.xml (hunk 1 above)
Comment (baranowb) [reply spotbugs-exclude.xml:703]: AFAIR lambdas trigger those rules.
